string[] log
string[] inputTypes
string test
---
ClusterPoint[] data
